### PR TITLE
SDKS-2354 Align device binding error handling with iOS

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket

https://bugster.forgerock.org/jira/browse/SDKS-2354

# Description

Align device binding error handling with iOS

# Definition of Done Checklist:

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Ensure backward compatibility.
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] Functional spec is written/updated.
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

### CODE REVIEW

**Review by jeyanthanperiyasamy — APPROVED** [review]
LGTM

File: forgerock-auth/src/androidTest/java/org/forgerock/android/auth/callback/DeviceBindingListAndUnbind.java [hunk 1]
```diff
@@ -166,14 +160,17 @@ public void onException(Exception e) {
 
     @Test
     public void testListAndDeleteKeys() throws ExecutionException, InterruptedException, IOException, ApiException {
+        bindDevice("bind-pin"); // Bind the device with APPLICATION_PIN
+        bindDevice("bind");     // Bind the device with NONE
```
Comment (witrisna) [inline forgerock-auth/src/androidTest/java/org/forgerock/android/auth/callback/DeviceBindingListAndUnbind.java:164]: Now binding App Pin Key and Keystore key will remove the previous key, so only one key will persist on the client side. I have changed the test to test only one key per person per device.

File: forgerock-auth/src/main/java/org/forgerock/android/auth/callback/DeviceBindingCallback.kt [hunk 2]
```diff
@@ -207,6 +213,7 @@ open class DeviceBindingCallback : AbstractCallback, Binding {
         try {
             val status: DeviceBindingStatus
             withTimeout(getDuration(timeout)) {
+                clearKeys(context, deviceAuthenticator)
```
Comment (witrisna) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/callback/DeviceBindingCallback.kt:216]: When start to bind, remove all keys.

**Review by spetrov — APPROVED** [review]
Looks good to me!

**Pull request merged**